test/logging/logging_test.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logging
+
+import (
+	"errors"
+	"reflect"
+	"testing"
+)
+
+type abc struct {
+	A int
+	b string
+	C *de
+	F func()
+}
+
+type de struct {
+	D string
+	e float64
+}
+
+const runFailingTests = false
+
+var someStruct abc
+
+func init() {
+	someStruct = abc{
+		A: 42,
+		b: "some string",
+		C: &de{
+			D: "hello world",
+			e: 72.3,
+		},
+		F: InitializeLogger,
+	}
+}
+
+func TestTLogger(legacy *testing.T) {
+	verbosity = 5
+	InitializeLogger()
+	t, cancel := NewTLogger(legacy)
+	defer cancel()
+
+	var blank interface{}
+	blank = &someStruct
+
+	t.V(6).Info("Should not be printed")
+	t.V(4).Info("Should be printed!")
+	t.Run("A-Nice-Subtest", func(ts *TLogger) {
+		ts.V(0).Info("This is pretty important; everyone needs to see it!",
+			"some pointer", blank,
+			"some number", 42.0)
+		t.Run("A-Nested-Subtest", func(ts *TLogger) {
+			ts.Parallel()
+			ts.V(1).Info("I am visible!")
+			ts.V(6).Info("I am invisible!")
+			t.Collect("collected_non_error", "I'm not an error!")
+		})
+		t.Run("A-2nd-Nested-Subtest", func(ts *TLogger) {
+			ts.Parallel()
+			ts.V(1).Info("I am visible!")
+			ts.V(6).Info("I am also invisible!")
+		})
+	})
+	if runFailingTests {
+		t.Run("Failing1", func(ts *TLogger) {
```

Comment:
Those tests are expcluded, due to `false` constant above.

Reply:
I will resolve because its outcome depends entirely on the conversation in the section about the const declaration.